From aca0c5da8d83e32c9b63de36f6e810ac3e795278 Mon Sep 17 00:00:00 2001
From: Ram Chandra Jangir <rjangir@codeaurora.org>
Date: Fri, 3 Jul 2020 18:32:57 +0530
Subject: [PATCH] skbuff_debug: make skbuff_debugobj_fixup function return bool
 instead of int

skb debug is failing in arm compilation as in debugobjects patches,
one of the commit* b1e4d9d8, which makes fixup functions return bool instead of int.
This change makes skbuff_debugobj_fixup functions return bool instead of int.

Change-Id: I384dffaacc92d5c45dcc295882951f84b4e5f9d3
Signed-off-by: Ram Chandra Jangir <rjangir@codeaurora.org>
---
 net/core/skbuff_debug.c | 8 --------
 1 file changed, 8 deletions(-)

--- a/net/core/skbuff_debug.c
+++ b/net/core/skbuff_debug.c
@@ -144,11 +144,7 @@ EXPORT_SYMBOL(skbuff_debugobj_print_skb)
  *	Called when an error is detected in the state machine for
  *	the objects
  */
-#if defined(CONFIG_ARM64)
 static bool skbuff_debugobj_fixup(void *addr, enum debug_obj_state state)
-#else
-static int skbuff_debugobj_fixup(void *addr, enum debug_obj_state state)
-#endif
 {
 	struct sk_buff *skb = (struct sk_buff *)addr;
 	ftrace_dump(DUMP_ALL);
@@ -156,11 +152,7 @@ static int skbuff_debugobj_fixup(void *a
 	     state, skb, skb->sum, skbuff_debugobj_sum(skb));
 	skb_recycler_notifier_send_event(SKB_RECYCLER_NOTIFIER_FSM, skb);
 
-#ifdef CONFIG_ARM64
 	return true;
-#else
-	return 0;
-#endif
 }
 
 static struct debug_obj_descr skbuff_debug_descr = {
